wordpress-mobile/WordPress-iOS
# Show loader indicator and errors on the new Reader detail

Fixes #14237

This PR adds a loader indicator that is shown while the content hasn't been retrieved and rendered. It also shows a message in case of a failure or if the user if without connectivity.

It also:

* Improves WebView rendering time: we make sure to load the CSS and the text before everything, just after that step the other things are loaded. This results in the content appearing very quickly so the user can start reading right away.
* Removes `autoplay` from audio and videos. The current `WKWebView` plays that automatically, which is very annoying.
* Points the CSS to https://wordpress.com/calypso/reader-mobile.css (per https://github.com/Automattic/wp-calypso/pull/43073)

| Loading  | Error | Connectivity |
| ------------- | ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/7040243/84278188-d2083f80-ab0a-11ea-9ab1-d81a198c56a5.png"> | <img src="https://user-images.githubusercontent.com/7040243/84278237-e0565b80-ab0a-11ea-9e45-8b343151f490.png"> | <img src="https://user-images.githubusercontent.com/7040243/84278270-e8ae9680-ab0a-11ea-9dcc-82166bc4cb4e.png"> |
 
### To test

**Make sure that you have overridden the new reader Feature Flag and it's not using the older Reader (Me -> App Settings -> Debug -> "Reader content displayed in a webview" should be ON)**

Please, test the screens above in:

* iPhone: light and dark mode
* iPad: light and dark mode

To generate the error screen you can try to access a post which you don't have permission or either change `renderPostAndBumpStats`:

```swift
    private func renderPostAndBumpStats() {
        view?.showError()
    }
```

PR submission checklist:
